zulipterminal/ui_tools/views.py

```diff
@@ -1646,12 +1665,68 @@ def create_link_buttons(
 
         return link_widgets, link_width
 
+    def create_code_snippet_buttons(
+        self, controller: Any, code_snippets: List[Tuple[str, List[Tuple[str, str]]]]
+    ) -> Tuple[List[Any], int]:
+        code_snippet_widgets = []
+        code_snippet_width = 0
+
+        for index, snippet in enumerate(code_snippets):
+            language, snippet_list = snippet
+            language = "" if language is None else language
+            display_code, copy_code = self.get_code_from_snippet(snippet_list)
+            if display_code:
+                code_snippet_width = max(
+                    code_snippet_width, len(max(display_code, key=len))
+                )
+                # if language == "":
+                #     display_code[-1] = list(display_code[-1])
+                #     display_code[-1][-1] = (
+                #         display_code[-1][-1][:-1]
+                #         if display_code[-1][-1][-1] == "\n"
+                #         else display_code
+                #     )
+                #     display_code[-1] = tuple(display_code[-1])
```

Comment:
We should remove commented code from before; you can split this into a later commit, in case we want to refer to it for a later feature, or remove it and we can refer to the original PR instead.

Reply:
I have deleted the commented code.